java-checks/src/test/files/checks/DeadStoreCheck.java

```diff
@@ -246,3 +247,82 @@ void assertStatement(boolean x) {
   }
 
 }
+
+class NoIssueOnInitializers {
+
+  // no issue if variable initializer is 'true' or 'false'
+  boolean testBoolean(boolean arg0) {
+    boolean b1 = true; // Compliant
+    b1 = false;        // Noncompliant
+    b1 = true;         // Noncompliant
+    b1 = arg0;
+
+    boolean b2 = false; // Compliant
+    b2 = true;          // Noncompliant
+    b2 = false;         // Noncompliant
+    b2 = arg0;
+
+    boolean b3 = arg0;  // Noncompliant
+    b3 = true;          // Noncompliant
+    b3 = false;         // Noncompliant
+    b3 = arg0;
+
+    return b1 && b2 && b3;
+  }
+
+  // no issue if initializer is 'null'
+  Object testNull(boolean b, Object o) {
+    Object o1 = null;  // Compliant
+    o1 = new Object(); // Noncompliant
+    o1 = null;         // Noncompliant
+    o1 = o;
+
+    Object o2 = o; // Noncompliant
+    o2 = null;
+
+    return b ? o1 : o2;
+  }
+
+  //no issue if initializer is the empty String
+  String testNull(String s) {
+    String s1 = ""; // Compliant
+    s1 = "yolo";    // Noncompliant
+    s1 = "hello";
+
+    String s2 = "world"; // Noncompliant
+    s2 = "moto";
+
+    return s1 + s2;
+  }
+
+  // no issue if variable initializer is '-1', '0', or '1'
+  int testIntLiterals(int i) {
+
+    int w = 42; // Noncompliant
+    w = 1;      // Noncompliant
+    w = 0;      // Noncompliant
+    w = -1;
+
+    int x = (0); // Compliant
+    x = 1;       // Noncompliant
+    x = 0;       // Noncompliant
+    x = i;
+
+    int y = 1; // Compliant
+    y = 0;     // Noncompliant
+    y = -1;    // Noncompliant
+    y = i;
+
+    int z1 = -(1); // Compliant
+    z1 = 0;        // Noncompliant
+    z1 = 1;        // Noncompliant
+    z1 = i;
+
+    int z2 = -1;   // Compliant
+    z2 = 0;        // Noncompliant
+    z2 = 1;        // Noncompliant
```

Comment:
I can understand the test of assigning 0 or 1 not in declaration once but why repeating it over and over ? this is noise.